variantlib/commands/generate_index_json.py

```diff
@@ -60,64 +74,24 @@ def generate_index_json(args: list[str]) -> None:
                 logger.warning("%s: no METADATA file found", wheel)
                 continue
 
-            if (
-                variant_hash := wheel_metadata.get(METADATA_VARIANT_HASH_HEADER)
-            ) is None:
-                logger.debug("%s: not a Wheel Variant. Skipping ...", wheel)
-                continue
-
+            # Only valid Wheel Variants need to be processed
             if (
                 variant_entries := wheel_metadata.get_all(
                     METADATA_VARIANT_PROPERTY_HEADER
                 )
             ) is None:
                 logger.warning(
-                    "%s: Variant-hash present but no Variant property", wheel
+                    "%(wheel)s: This wheel does not declare any `%(key)s`",
+                    {"wheel": wheel, "key": METADATA_VARIANT_PROPERTY_HEADER},
                 )
                 continue
 
-            if (
-                wheel_providers := wheel_metadata.get_all(
-                    METADATA_VARIANT_PROVIDER_HEADER
-                )
-            ) is None:
-                logger.info("%s: no Variant-provider", wheel)
-
-            else:
-                for wheel_provider in wheel_providers:
-                    ns, _, provider = wheel_provider.partition(",")
-                    ns = ns.strip()
-                    provider = provider.strip()
-                    if not ns or not provider:
-                        raise ValueError(
-                            f"{wheel}: Invalid {METADATA_VARIANT_PROVIDER_HEADER}: "
-                            f"{wheel_provider}"
-                        )
-                    if providers.setdefault(ns, provider) != provider:
-                        raise KeyError(
-                            f"Conflicting providers found for namespace {ns}: "
-                            f"{providers[ns]} and {provider}"
-                        )
+            # ============== Variant Properties Processing ================ #
 
             vprops = [VariantProperty.from_str(vprop) for vprop in variant_entries]
-            for vprop in vprops:
-                known_namespaces.add(vprop.namespace)
 
             try:
                 vdesc = VariantDescription(vprops)
-                if vdesc.hexdigest != variant_hash:
-                    logger.error(
-                        "%(wheel)s has been rejected due to a variant hash mismatch: "
```

Comment:
I removed this part - we don't "care" in this command if the hashes matches / mismatches.
Especially that the value inside METADATA is purely informative and does have any direct use.

Let's just compute the hash on the fly and that's it.

This command is "pre-computing what variant is what hash" - this logic may force us to skip a variant that the installer could have picked up.